src/SharpBrick.PoweredUp.Mobile/XamarinPoweredUpBluetoothAdapter.cs

```diff
@@ -0,0 +1,142 @@
+                {
+                    var data = advertisementRecord.Data.ToList();
+                    data.RemoveRange(0, 2);
+                    info.ManufacturerData = data.ToArray();
+                    info.Name = args.Device.Name;
+                    info.BluetoothAddress = _deviceInfo.GetNativeDeviceInfo(args.Device.NativeDevice).MacAddressNumeric;
+
+                    AddInternalDevice(args.Device, info);
+                        await discoveryHandler(info).ConfigureAwait(false);
+                }
+            }
+        }
+
+        private void AddInternalDevice(IDevice device, PoweredUpBluetoothDeviceInfo info)
+        {
+            if (!_discoveredDevices.ContainsKey(info.BluetoothAddress))
+            {
+                _discoveredDevices.Add(info.BluetoothAddress, device);
+            }
+            else
+            {
+                _discoveredDevices[info.BluetoothAddress] = device;
+            }
+        }
+
+        private bool DeviceFilter(IDevice arg)
+        {
+            if (arg == null) return false;
+
+            System.Diagnostics.Debug.WriteLine(arg.Name);
+
+            var manufacturerData = arg.AdvertisementRecords.FirstOrDefault(x => x.Type == AdvertisementRecordType.ManufacturerSpecificData);
+
+            if (manufacturerData?.Data == null || manufacturerData.Data.Length < 8) return false;
+
+            // https://lego.github.io/lego-ble-wireless-protocol-docs/index.html#advertising
+            // Length and Data Type Name seems to be already trimmed away
+            // Manufacturer ID should be 0x0397 but seems in little endian encoding. I found no notice for this in the documentation except in version number encoding
+
+            switch (manufacturerData.Data[3])
+            {
+                case 0x00:
+                    System.Diagnostics.Debug.WriteLine("System: LEGO Wedo 2.0, Device: WeDo Hub");
+                    break;
+                case 0x20:
+                    System.Diagnostics.Debug.WriteLine("System: LEGO Duplo, Device: Duplo Train");
+                    break;
+                case 0x40:
+                    System.Diagnostics.Debug.WriteLine("System: System, Device: Boost Hub");
+                    break;
+                case 0x41:
+                    System.Diagnostics.Debug.WriteLine("System: System, Device: 2 Port Hub");
+                    break;
+                case 0x42:
+                    System.Diagnostics.Debug.WriteLine("System: System, Device: 2 Port Handset");
+                    break;
+                default:
+                    if (manufacturerData.Data[3] >= 96 && manufacturerData.Data[3] < 128)
+                    {
+                        System.Diagnostics.Debug.WriteLine("System: LEGO System, Device: Currently unknown");
+                    }
+                    break;
+            }
+
+            return manufacturerData.Data[0] == 0x97 || manufacturerData.Data[1] == 0x03;

```

Comment:
As mentioned in https://lego.github.io/lego-ble-wireless-protocol-docs/index.html#advertising the Manufacturer specific data starts with 0x0397. 

That's the strange part. When I look into WinRTPoweredUpBluetoothAdapter I can see, that those two bytes are missing. 
Here i'll check in the DeviceFilter, that those two bytes are set and only BLE devices with this specific bytes are encountered for the DiscoveredDevice event. (The or must be an and)
I also need to remove those two bytes in the revceived handler otherwise the discoveryHandler() would throw an exception

Reply:
Yeah true. There was something strange about it. Good that this adapter gets it.